ogr/services/github/service.py

```diff
@@ -71,13 +73,25 @@ def __init__(
         self.read_only = read_only
         self.authentication = github_authentication
 
+        if isinstance(max_retries, Retry):
+            self._max_retries = max_retries
+        else:
+            self._max_retries = Retry(
+                total=int(max_retries),
+                read=0,
+                method_whitelist=["DELETE", "GET", "PATCH", "POST", "PUT"],
+                status_forcelist=[500, 403, 401],
```

Comment:
403, 500 - definitely agree, ideally all variants of 500 (especially 503)

for 401 - what exactly is the use case, are we actually getting 401 somewhere which succeeds moments later?

Reply:
> for 401 - what exactly is the use case, are we actually getting 401 somewhere which succeeds moments later?

It was [requested](https://github.com/packit/packit-service/issues/830) to retry on `BadCredentialsException`. It is [raised when HTTP status is 401](https://github.com/PyGithub/PyGithub/blob/6900f187b53711d7e9e43fcadede2e44d7d12f7d/github/Requester.py#L412). AFAIK it shouldn't break [HTTP authentication](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication) (but of course the retry mechanism will slow it down).

Reply:
understood and makes sense, let's also keep 401